bbq/compiler/desugar.go

```diff
@@ -633,10 +556,15 @@ func (d *Desugar) VisitCompositeDeclaration(declaration *ast.CompositeDeclaratio
 
 	// Recursively de-sugar nested declarations (functions, types, etc.)
 
-	prevInheritedFuncs := d.inheritedFuncsWithConditions
+	prevInheritedFuncsWithConditions := d.inheritedFuncsWithConditions
+	prevEnclosingInterfaceType := d.enclosingInterfaceType
+
 	d.inheritedFuncsWithConditions = d.inheritedFunctionsWithConditions(compositeType)
+	d.enclosingInterfaceType = nil
+
 	defer func() {
-		d.inheritedFuncsWithConditions = prevInheritedFuncs
+		d.inheritedFuncsWithConditions = prevInheritedFuncsWithConditions
+		d.enclosingInterfaceType = prevEnclosingInterfaceType
```

Comment:
Also an unrelated fix, but needed for the FT test. Composite types can be inside interfaces, in the form of events. And events have initializer functions, that should be desugar-ed/compiled. This path was uncovered by the desugar-ing of initializers (fix) above.

Reply:
Same as above: Is this fix already tested?